types/recharts/index.d.ts

```diff
@@ -920,20 +921,18 @@ export interface LabelProps extends Partial<PresentationAttributes> {
 
 export class LabelList extends React.Component<LabelListProps> { }
 
-export interface LabelListProps {
+export type LabelListProps = {
     angle?: number;
     children?: React.ReactNode[] | React.ReactNode;
     className?: string;
     clockWise?: boolean;
     content?: React.ReactElement | ContentRenderer<LabelProps>;
     data?: number;
-    dataKey: string | number | RechartsFunction;
     formatter?: LabelFormatter;
     id?: string;
     offset?: number;
     position?: PositionType;
-    valueAccessor?: RechartsFunction;
-}
+} & ({ dataKey: string | number | RechartsFunction } | { valueAccessor: RechartsFunction });
```

Comment:
If I understood correctly, you want to have exclusive OR here, but with this definition TypeScript will merge both of those types and developer will be able to pass both properties without any errors 🤷‍♂ . You can prevent this by setting other property to `never` type like so:

```suggestion
} & ({ dataKey: string | number | RechartsFunction, valueAccessor?: never } | { valueAccessor: RechartsFunction, dataKey?: never });
```

See [TypeScript playground](https://www.typescriptlang.org/play/#code/KYDwDg9gTgLgBDAnmYcAyBDARsANmgSwGcYAFKCMIuAXjgG8AoOFuDAOwHNdgB+ALjjsArgFscUANzNWAY1wYiRAHIZRfQSSgEu01nHkRZAawDqxDXCwQIPDntYATDDAwChYiQ5YFH7rTqc3nAQAGahRMAw7iLiwFKMAL5wAGRwABT0cM6uANLAiJow2lxwAD4ecVBwyRVZAG4YuMLAAIKyssBK0ILpAJS0AHxw9RC+NX3SjAD003Cm0MbUoTqoWMLwRAAWEMK4jnBQUcJQ7Gxn8RRQjLIQ7CRwYBRUgpg4+MRkz9R0TPoc3GAggATABmAA0MhYOQw+UKcAA5DAujAEZD9I1mm0Ol0iD0MgMaMNhOxHMAVuxgI4koxGEgUPMCDAdhsADwAFXBcAAqsNfnAANqkOA6OAAURA8mEZNZxgKYTgnLgcsQCt5AF0YsB6vEatJ6agABoAeQAShyubzaBl2eUeQNQMjSdQIFgAFbAWTwXgZczM3YwC084ZpbkDCrpP0swPcrnskOKgaCW0VblTUCQWAIZCoN54QgkciUIgm03Wv6sAE8GKeeLBeSKFRqSwBXRQgy4IxmCzuay2YD2dswmtVYK+fzFQLBMIRKIjrxJVJwUusrIwuFFEqcO2xCQ1LkNJotdqdbpQXqE4ajcaJQZTWbiqBXLkAd0W1EUcAznuR1Nu93gEBoCLF50GwfNPhAkszXLdsqyBOAwXRJwXFhApBCRFE0XbTFjxxM8LyGOASTJCkqRpIA)(I've used custom XOR helper here though, but it will work with `never` as well)

Reply:
Thanks for your review. You are right, both properties shouldn't be set together. I just assumed dataKey will take precedence, but this can be confusing.
So your suggestion is definitely correct.

I tested it too and it works fine.